addons/mail/static/src/discuss/core/web/discuss_sidebar_categories_patch.js

```diff
@@ -77,6 +86,40 @@ const DiscussSidebarCategoryPatch = {
             });
         }
     },
+    renameCategory(name) {
+        this.category.name = name;
+        this.state.renaming = false;
+        this.env.services.orm.write("discuss.channel.category", [this.category.id], { name });
+    },
+    async leaveChannels() {
+        this.dialogService.add(ConfirmationDialog, {
+            body: _t(
+                "You are going to leave all channels in the category %s. Are you sure you want to continue?",
+                this.category.name
+            ),
+            confirmLabel: _t("Leave Channels"),
+            confirm: () => {
+                for (const thread of this.category.threads) {
+                    thread.leave();
+                }
+            },
+            cancel: () => {},
+        });
+    },
+    deleteCategory() {
+        this.dialogService.add(ConfirmationDialog, {
+            body: _t(
+                "You are going to delete the category %s. Are you sure you want to continue?",
```

Comment:
I would remove these too complex buttons in sidebar. We can customise the category of a channel in the settings, that's all we need.

Reply:
Ok, specs should be updated then